pkg/network/buffer.go

```diff
@@ -0,0 +1,70 @@
+package network
+
+import "math"
+
+const minBufferSize = 256
+
+// Buffer encapsulates a resizing buffer for providing ConnectionStat pointers
+// to read from the windows driver
+type Buffer struct {
```

Comment:
nit: `Buffer` is a bit of a generic name. Maybe something like `ConnectionBuffer`?